lib/schema/Schemas.js

```diff
@@ -0,0 +1,396 @@
+                    "errorMessage": {
+                        "type": "must be a string"
+                    }
+                },
+                "descriptionDefault": {
+                    "type": "string",
+                    "errorMessage": {
+                        "type": "must be a string"
+                    }
+                },
+                "defaultWindowStyle": {
+                    "type": "object",
+                    "properties": {
+                        "width": {
+                            "type": "integer",
+                            "errorMessage": {
+                                "type": "must be an integer"
+                            }
+                        },
+                        "height": {
+                            "type": "integer",
+                            "errorMessage": {
+                                "type": "must be an integer"
+                            }
+                        },
+                        "x": {
+                            "type": "integer",
+                            "errorMessage": {
+                                "type": "must be an integer"
+                            }
+                        },
+                        "y": {
+                            "type": "integer",
+                            "errorMessage": {
+                                "type": "must be an integer"
+                            }
+                        }
+                    },
+                    "errorMessage": {
+                        "type": "must be an object"
+                    }
+                }
+            },
+            "errorMessage": {
+                "type": "must be an object"
+            }
+        }
+    },
+    "$schema": "http://json-schema.org/schema#",
+    "type": "object",
+    "if": {
+        "properties": {
+            "pluginType": {
+                "const": "library"
+            }
+        }
+    },
+    "then": {
+        "required": ["libraryName", "libraryVersion"]
+    },
+    "else": {
+        "if": {
+            "properties": {
+                "pluginType": {
+                    "const": "nodeAuthentication"
+                }
+            }
+        },
+        "then": {
+            "required": ["authenticationCategory", "filename"]
+        },
+        "else": {
+            "if": {
+                "properties": {
+                    "pluginType": {
+                        "const": "desktop"
+                    }
+                }
+            },
+            "then": {
+                "properties": {
+                    "webContent": {
+                        "if": {
+                            "additionalProperties": false //Checks if the webcontent is empty
+                        },
+                        "then": {},
+                        "else": {
+                            "allOf": [
+                                { "$ref": "#/definitions/webContent" },
```

Comment:
A previous requirement of this work was that the definitions themselves be JSON, and here's why: if we run into a situation where someone needs a plugin that uses a schema version that the server doesnt have, you can provide a JSON file harmlessly. But, sending someone a JS file can be a security concern for many. 

Either way, what must be done is that each schema must be in its own JSON/JS file, such that we can flexibly validate revisions to the spec for the plugins that will use it.